File: archive/scripts_legacy/db_inspect.js
const Database = require('better-sqlite3');
const path = require('path');
const db = new Database('c:/Users/HAMDI/Desktop/HamdiProno/stitch/data/tactical.db');
const tables = db.prepare("SELECT name FROM sqlite_master WHERE type='table'").all();
console.log('Tables:', tables);

for (const table of tables) {
    const schema = db.prepare(`PRAGMA table_info(${table.name})`).all();
    console.log(`Schema for ${table.name}:`, schema);
    const sample = db.prepare(`SELECT * FROM ${table.name} LIMIT 1`).get();
    console.log(`Sample for ${table.name}:`, sample);
}

File: tests/db_inspect.py
import sqlite3
conn = sqlite3.connect('data/tactical.db')

# List tables
tables = conn.execute("SELECT name FROM sqlite_master WHERE type='table'").fetchall()
print("TABLES:", [t[0] for t in tables])

# Check archive_matches team names for national teams
try:
    rows = conn.execute("SELECT DISTINCT homeTeam FROM archive_matches LIMIT 50").fetchall()
    print("\narchive_matches homeTeam samples:")
    for r in rows: print(f"  {r[0]}")
except: print("No archive_matches")

# Check archive_football_data
try:
    rows = conn.execute("SELECT DISTINCT home_team FROM archive_football_data LIMIT 50").fetchall()
    print("\narchive_football_data home_team samples:")
    for r in rows: print(f"  {r[0]}")
except: print("No archive_football_data")

# Check international_results
try:
    rows = conn.execute("SELECT DISTINCT home_team FROM international_results LIMIT 50").fetchall()
    print("\ninternational_results home_team samples:")
    for r in rows: print(f"  {r[0]}")
except: print("No international_results")

# Check matches table columns
try:
    cols = conn.execute("PRAGMA table_info(matches)").fetchall()
    print("\nmatches columns:", [c[1] for c in cols])
except: pass
